common/djangoapps/student/models.py

```diff
@@ -1935,12 +1970,12 @@ def refund_cutoff_date(self):
                     "Order={number} and user {user}".format(number=order_number, user=self.user.id))
                 return None
 
-        refund_window_start_date = max(
-            datetime.strptime(date_placed, ECOMMERCE_DATE_FORMAT),
-            self.course_overview.start.replace(tzinfo=None)
-        )
-
-        return refund_window_start_date.replace(tzinfo=UTC) + EnrollmentRefundConfiguration.current().refund_window
```

Comment:
same question as above?

Reply:
Adding original snippet of `refund_cutoff_date` here to clear the confusion:

```python
    def refund_cutoff_date(self):
        """ Calculate and return the refund window end date. """
        # NOTE: This is here to avoid circular references
        from openedx.core.djangoapps.commerce.utils import ecommerce_api_client, ECOMMERCE_DATE_FORMAT

        date_placed = self.get_order_attribute_value('date_placed')

        if not date_placed:
            order_number = self.get_order_attribute_value('order_number')
            if not order_number:
                return None

            try:
                order = ecommerce_api_client(self.user).orders(order_number).get()
                date_placed = order['date_placed']
                # also save the attribute so that we don't need to call ecommerce again.
                username = self.user.username
                enrollment_attributes = get_enrollment_attributes(username, str(self.course_id))
                enrollment_attributes.append(
                    {
                        "namespace": "order",
                        "name": "date_placed",
                        "value": date_placed,
                    }
                )
                set_enrollment_attributes(username, str(self.course_id), enrollment_attributes)
            except HttpClientError:
                log.warning(
                    "Encountered HttpClientError while getting order details from ecommerce. "
                    "Order={number} and user {user}".format(number=order_number, user=self.user.id))
                return None

            except HttpServerError:
                log.warning(
                    "Encountered HttpServerError while getting order details from ecommerce. "
                    "Order={number} and user {user}".format(number=order_number, user=self.user.id))
                return None

            except SlumberBaseException:
                log.warning(
                    "Encountered an error while getting order details from ecommerce. "
                    "Order={number} and user {user}".format(number=order_number, user=self.user.id))
                return None

        refund_window_start_date = max(
            datetime.strptime(date_placed, ECOMMERCE_DATE_FORMAT),
            self.course_overview.start.replace(tzinfo=None)
        )

        return refund_window_start_date.replace(tzinfo=UTC) + EnrollmentRefundConfiguration.current().refund_window
```